ui/page/dcrdex/market.go

```diff
@@ -0,0 +1,1051 @@
+		return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+			layout.Rigid(func(gtx C) D {
+				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						if baseIcon == nil {
+							return D{}
+						}
+						return baseIcon.Layout20dp(gtx)
+					}),
+					layout.Rigid(func(gtx C) D {
+						return layout.Inset{Right: u2, Left: u2}.Layout(gtx, pg.Theme.Label(values.TextSize16, baseAsset.String()).Layout)
+					}),
+				)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Right: u2, Left: u2}.Layout(gtx, pg.Theme.Label(values.TextSize16, "/").Layout)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						if quoteIcon == nil {
+							return D{}
+						}
+						return quoteIcon.Layout20dp(gtx)
+					}),
+					layout.Rigid(func(gtx C) D {
+						return layout.Inset{Right: u2, Left: u2}.Layout(gtx, pg.Theme.Label(values.TextSize16, quoteAsset.String()).Layout)
+					}),
+				)
+			}),
+		)
+	}
+}
+
+// OnNavigatedFrom is called when the page is about to be removed from
+// the displayed window. This method should ideally be used to disable
+// features that are irrelevant when the page is NOT displayed.
+// NOTE: The page may be re-displayed on the app's window, in which case
+// OnNavigatedTo() will be called again. This method should not destroy UI
+// components unless they'll be recreated in the OnNavigatedTo() method.
+// Part of the load.Page interface.
+func (pg *DEXMarketPage) OnNavigatedFrom() {}
+
+// Layout draws the page UI components into the provided C
+// to be eventually drawn on screen.
+// Part of the load.Page interface.
+func (pg *DEXMarketPage) Layout(gtx C) D {
+	return pg.pageContentLayout(gtx)
+}
+
+func (pg *DEXMarketPage) pageContentLayout(gtx C) D {
+	pageContent := []func(gtx C) D{
+		pg.priceAndVolumeDetail,
+		pg.orderFormAndOrderBook,
+		pg.openOrdersAndHistory,
+	}
+
+	return cryptomaterial.LinearLayout{
+		Width:     cryptomaterial.MatchParent,
+		Height:    cryptomaterial.MatchParent,
+		Direction: layout.Center,
+	}.Layout2(gtx, func(gtx C) D {
+		return cryptomaterial.LinearLayout{
+			Width:  gtx.Dp(values.AppWidth - 50),
+			Height: cryptomaterial.MatchParent,
+			Margin: layout.Inset{
+				Bottom: values.MarginPadding30,
+				Right:  u10,
+				Left:   u10,
+			},
+		}.Layout2(gtx, func(gtx C) D {
+			return pg.Theme.List(pg.scrollContainer).Layout(gtx, 1, func(gtx C, index int) D {
+				return layout.Stack{}.Layout(gtx,
+					layout.Expanded(func(gtx C) D {
+						return layout.Inset{Top: 110}.Layout(gtx, func(gtx C) D {
+							return pg.pageContainer.Layout(gtx, len(pageContent), func(gtx C, i int) D {
+								return pageContent[i](gtx)
+							})
+						})
+					}),
+					layout.Stacked(pg.serverAndCurrencySelection()),
+				)
+			})
+		})
+	})
+}
+
+func (pg DEXMarketPage) addServerDisplay() func(gtx C) D {
+	return func(gtx cryptomaterial.C) cryptomaterial.D {
```

Comment:
Hmm. You can still prevent it from being selectable if included as a normal item using a `item.preventSelect bool` property. You only set to true for items like this that should not be selectable. Should result in fewer codes, I think; and an easier to understand flow. Took me a while to realize that this "extraDisplay" actually shows up in a similar way to the other dropdown items. I expect others will have the same issue.

Reply:
Actually, it can be selectable. When it is selected, you start the process of adding a server. If that process doesn't complete, you clear the selection. If it completes, you select the newly added server. The UI won't show the "Add Server" selection for long, if at all. If it does show for a brief moment, it's understandable. The user was supposed to select a server and he selected the option to add a new server. It makes sense if that selection is displayed briefly as long as the user's request to add a new server is honored.

Reply:
Re: https://github.com/crypto-power/cryptopower/pull/208#discussion_r1405894480, done.